src/libs/Navigation/Navigation.js

```diff
@@ -48,37 +49,55 @@ function setDidTapNotification() {
  * @returns {Boolean}
  */
 function canNavigate(methodName, params = {}) {
```

Comment:
```suggestion
function canNavigate(methodName, params = {}) {
    return new Promise((resolve) => {
        if (navigationRef.isReady() && !isNavigating) {
            resolve(true);
        } else if (isNavigating) {
            Log.hmmm(`[Navigation] ${methodName} failed because navigation is progress`, params);
            resolve(false);
        } else {
            Log.hmmm(`[Navigation] ${methodName} failed because navigation ref was not yet ready`, params);
            resolve(false);
        }
    });
}
```

Reply:
I think we can simplify `canNavigate` a bit

Reply:
Done